### DISCUSSION TITLE
How to get entity from raycast?

### DISCUSSION THREAD

**Discussion by JacktheFabricRipper** [pokepetter/ursina#304] [Q&A]
I used raycast.entity but it just returned render/scene/entity. what am I doing wrong?

**Comment by JacktheFabricRipper** [comment]
This even happens when the ray hasn't hit anything

**Reply by pokepetter** [reply]
raycast.entity or raycast(...).entity?
The thing you posted is an Entity, so seems correct.

How did you get it when it didn't anything though? Got an example?

**Comment by leer-ursina** [comment]
hello,
try this with raycast example:
[https://www.ursinaengine.org/collision.html](url)

set a name to object possibly hitted
wall_left = Entity(name="wall_left", ........

after the bolck code:
if not hit_info.hit:
            self.position += self.direction * 5 * time.dt
add:
else:print(hit_info.entity.name)
or
else:print(hit_info.entities)

it uses the ursina.HitInfo
HitInfo doc at ursina cheat sheet
